addons/mail/static/src/models/thread_cache/thread_cache.js

```diff
@@ -313,15 +332,14 @@ function factory(dependencies) {
          *
          * @private
          */
-        _onHasToLoadMessagesChanged() {
+        async _onHasToLoadMessagesChanged() {
             if (!this.hasToLoadMessages) {
                 return;
             }
-            this._loadMessages().then(fetchedMessages => {
-                for (const threadView of this.threadViews) {
-                    threadView.addComponentHint('messages-loaded', { fetchedMessages });
-                }
-            });
+            const fetchedMessages = await this._loadMessages();
+            for (const threadView of this.threadViews) {
+                threadView.addComponentHint('messages-loaded', { fetchedMessages });
+            }
```

Comment:
I'm not sure if this change is acceptable. Doesn't it make the update cycle become async? Like if there are other computes following this one, as they still made synchronously compared to this one?

Reply:
It doesn't change much. This only makes returning promise explicit (and use `async/await` instead of `.then()`), but the async behaviour was already there before.

Reply:
But the previous method didn't return the promise. But I guess that doesn't change anything if we don't wait it.

Reply:
should use `this.async` btw